packages/vue-app/template/client.js

```diff
@@ -598,42 +653,62 @@ function addHotReload($component, depth) {
       next: next.bind(this)
     })
     const context = app.context
+
     <% if (loading) { %>
-    if (this.$loading.start && !this.$loading.manual) this.$loading.start()
+    if (this.$loading.start && !this.$loading.manual) {
+      this.$loading.start()
+    }
     <% } %>
+
     callMiddleware.call(this, Components, context)
     .then(() => {
+      <% if (features.layouts) { %>
       // If layout changed
-      if (depth !== 0) return Promise.resolve()
+      if (depth !== 0) {
+        return true
```

Comment:
Just a heads-up about this change (and some more below), returning in a thenable chain should always result in a promise.

Try eg: 
```js
new Promise(r => setTimeout(r, 1000)).then(() => true).then((v) => console.log('Done', v))
```